### PR TITLE
CP-12641 and CP-12475: XenCenter work for thin provisioning

### PR DESCRIPTION
**Pull request by GaborApatiNagy**
-New SR wizard: new page
-Add Virtual Disk dialog: new fields

Signed-off-by: Gabor Apati-Nagy <gabor.apati-nagy@citrix.com>

### CODE REVIEW

**Comment by GaborApatiNagy** [comment]
I am expecting one more commit before this is ready for review. (Realised that some more change is needed in SR Wizard for HBA option...)

File: XenAdmin/Controls/SrPicker.cs [hunk 1]
```diff
@@ -199,6 +208,24 @@ public void SetAffinity(Host host)
...
+        /// <summary>
+        /// Returns how much disk space is required to create the disk on an SR
+        /// This depends on whether the SR is thin provisioned and what the initial allocation rate is
+        /// </summary>
+        /// <param name="sr"></param>
+        /// <returns></returns>
+        private long GetRequiredDiskSizeForSR(SR sr)
+        {
+            if (sr != null && sr.sm_config != null
+                && sr.sm_config.ContainsKey("allocation") && sr.sm_config["allocation"] == "dynamic")
```
Comment (MihaelaStoica) [inline XenAdmin/Controls/SrPicker.cs]: The check that SR is thinly provisioned can be extracted in a  SR read-only property, as it is used in many places.

**Comment by GaborApatiNagy** [comment]
Fixed that, works as expected.

**Comment by MihaelaStoica** [comment]
Looks good. Merging.

**Pull request merged**